### PR TITLE
feat: add `bundleStrategy` option

### PR DESCRIPTION
**Pull request by paoloricciuti**
This adapts the code from @benmccann branch https://github.com/sveltejs/kit/tree/iife to the new shape of the repo. I also had to make some other changes namely:

1. instead of using the `iife` output (which doesn't export anything and put everything in a global variable) i used `manualChunks` with a single chunk
2. since we have everything in a single file i "namespaced" `app` by `export * as app`
3. on the client when resolving the bundle i've assigned `kit.app` as a fallback for `app`

I'm not super-familiar with the test suite so if you can suggest how and were i can test this (i guess it should be a new app?) i would appreciate (also not really sure if this has any repercussions).

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

**Comment by Rich-Harris** [comment]
preview: https://svelte-dev-git-preview-kit-13173-svelte.vercel.app/

_this is an automated message_

File: packages/kit/src/exports/vite/index.js [hunk 1]
```diff
@@ -606,14 +609,17 @@ async function kit({ svelte_config }) {
 						input[name] = path.resolve(file);
 					});
 				} else {
```
Comment (eltigerchino) [inline packages/kit/src/exports/vite/index.js:611]:
```suggestion
				} else if (!svelte_config.kit.output.codeSplit) {
```

File: packages/kit/src/exports/vite/index.js [hunk 2]
```diff
@@ -606,14 +609,17 @@ async function kit({ svelte_config }) {
...
+						input['entry/bundle'] = `${runtime_directory}/client/bundled-entry.js`;
+					} else {
+						input['entry/start'] = `${runtime_directory}/client/entry.js`;
+						input['entry/app'] = `${kit.outDir}/generated/client-optimized/app.js`;
+						manifest_data.nodes.forEach((node, i) => {
+							if (node.component || node.universal) {
+								input[`nodes/${i}`] = `${kit.outDir}/generated/client-optimized/nodes/${i}.js`;
+							}
+						});
+					}
```
Comment (eltigerchino) [inline packages/kit/src/exports/vite/index.js:622]:
```suggestion
						input['entry/bundle'] = `${runtime_directory}/client/bundled-entry.js`;
					} else {
						input['entry/start'] = `${runtime_directory}/client/entry.js`;
						input['entry/app'] = `${kit.outDir}/generated/client-optimized/app.js`;
						manifest_data.nodes.forEach((node, i) => {
							if (node.component || node.universal) {
								input[`nodes/${i}`] = `${kit.outDir}/generated/client-optimized/nodes/${i}.js`;
							}
						});
```

File: packages/kit/src/exports/vite/index.js (hunk 1 above)
Comment (paoloricciuti) [reply packages/kit/src/exports/vite/index.js:613]: thanks...applied in the last commit

**Comment by Rich-Harris** [comment]
Going to change this to `bundleStrategy: 'split' | 'single'` rather than `codeSplit: boolean`, so that we can accommodate more options in future

File: .changeset/beige-carpets-wave.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': minor
+---
+
+feat: add `codeSplit` option
```
Comment (Rich-Harris) [inline .changeset/beige-carpets-wave.md:5]:
```suggestion
feat: add `bundleStrategy: 'split' | 'single'` option
```

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**